core/common/result.hpp

```diff
@@ -8,17 +8,18 @@
 
 #include <ciso646>
 
-#include <optional>
 #include <boost/variant.hpp>
+#include <optional>
```

Comment:
Why? According to [google code style](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) other libraries should follow after C++ standard libraries

Reply:
I removed this header